File: fs_find-dirty-series-names.php
<?php

function startsWith( $haystack, $needle ) {
    $length = strlen( $needle );
    return substr( strtolower($haystack), 0, $length ) === strtolower($needle);
}

function endsWith( $haystack, $needle ) {
    $length = strlen( $needle );
    if( !$length ) {
        return true;
    }
    return substr( $haystack, -$length ) === $needle;
}

// recursive directory scan
function recursiveScan($dir) {
    $tree = glob(rtrim($dir, '/') . '/*');
    if (is_array($tree)) {
        foreach($tree as $file) {
            if (is_dir($file)) {
                //echo $file . "\n";
                recursiveScan($file);
            } elseif (is_file($file)) {
                if(endsWith($file, "poster.jpg")) continue;
                if(endsWith($file, "folder.jpg")) continue;
                if(endsWith($file, "banner.jpg")) continue;
                if(endsWith($file, "fanart.jpg")) continue;
                if(endsWith($file, "logo.jpg")) continue;
                if(endsWith($file, "landscape.jpg")) continue;
                if(endsWith($file, "clearart.png")) continue;
                if(endsWith($file, "logo.png")) continue;
                if(endsWith($file, "tvshow.nfo")) continue;

                $parts = explode("/", $file);

                $correct_series_name = startsWith($parts[count($parts) -1], $parts[count($parts) -3]);
                $correct_series_name_nospaces = startsWith($parts[count($parts) -1], str_replace(" ", "", $parts[count($parts) -3]));

                if( $correct_series_name == false && $correct_series_name_nospaces == false) {
                    echo $file . "\n";
                }
                else {
                    //echo ".";
                }
            }
        }
    }
}


function Scan($dir) {
    $tree = glob(rtrim($dir, '/') . '/*');
    if (is_array($tree)) {
        foreach($tree as $file) {
            //echo $file . "\n";
            recursiveScan($file);
        }
    }
}

Scan("/series");

File: fs_find-dirty-series-names.py
#!/usr/bin/env python3
import os
import glob

def startsWith(haystack, needle):
    length = len(needle)
    return haystack.lower()[:length] == needle.lower()

def endsWith(haystack, needle):
    length = len(needle)
    if not length:
        return True
    return haystack[-length:] == needle

def recursiveScan(dir):
    tree = glob.glob(dir.rstrip('/') + '/*')
    if isinstance(tree, list):
        for file in tree:
            if os.path.isdir(file):
                recursiveScan(file)
            elif os.path.isfile(file):
                if endsWith(file, "poster.jpg"): continue
                if endsWith(file, "folder.jpg"): continue
                if endsWith(file, "banner.jpg"): continue
                if endsWith(file, "fanart.jpg"): continue
                if endsWith(file, "logo.jpg"): continue
                if endsWith(file, "landscape.jpg"): continue
                if endsWith(file, "clearart.png"): continue
                if endsWith(file, "logo.png"): continue
                if endsWith(file, "tvshow.nfo"): continue

                parts = file.split("/")

                correct_series_name = startsWith(parts[-1], parts[-3])
                correct_series_name_nospaces = startsWith(parts[-1], parts[-3].replace(" ", ""))

                if not (correct_series_name or correct_series_name_nospaces):
                    print(file)
                else:
                    pass

def Scan(dir):
    tree = glob.glob(dir.rstrip('/') + '/*')
    if isinstance(tree, list):
        for file in tree:
            recursiveScan(file)

Scan("/series")
